### PR TITLE
Fix parameters not compatible with Solaris

### PR DESCRIPTION
**Pull request by Phil-Friderici**
Suggestion for a fix of issue https://github.com/ghoneycutt/puppet-module-ssh/issues/208

### CODE REVIEW

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -971,11 +971,15 @@
...
 
   describe 'sshd_config_permittunnel param' do
-    ['yes','point-to-point','ethernet','no'].each do |value|
+    ['yes','point-to-point','ethernet','no','unset'].each do |value|
       context "set to #{value}" do
         let (:params) { { :sshd_config_permittunnel => value } }
 
-        it { should contain_file('sshd_config').with_content(/^PermitTunnel #{value}$/) }
+        if value == 'unset'
+          it { should contain_file('sshd_config').without_content(/^PermitTunnel/) }
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: negative matches should have `/^\s*` as the start of the regex to protect us from whitespace shifting in the template.

File: templates/sshd_config.erb [hunk 2]
```diff
@@ -166,7 +166,9 @@ X11UseLocalhost <%= @sshd_x11_use_localhost %>
 PrintMotd <%= @sshd_config_print_motd %>
 #PrintLastLog yes
 #TCPKeepAlive yes
-TCPKeepAlive <%= @sshd_config_tcp_keepalive %>
+<% if @sshd_config_tcp_keepalive_real =~ /(yes|no)/ -%>
```
Comment (ghoneycutt) [inline templates/sshd_config.erb]: Would `!= 'unset'` make this easier to change in the future as valid options to the parameter change?

File: templates/sshd_config.erb [hunk 3]
```diff
@@ -195,7 +197,9 @@ MaxSessions <%= @sshd_config_maxsessions %>
 <% end -%>
 
 #PermitTunnel no
-PermitTunnel <%= @sshd_config_permittunnel %>
+<% if @sshd_config_permittunnel_real =~ /(yes|no|point-to-point|ethernet)/ -%>
```
Comment (ghoneycutt) [inline templates/sshd_config.erb]: Would `!= 'unset'` make this easier to change in the future as valid options to the parameter change?

**Comment by ghoneycutt** [comment]
I'm taking it that the other parameters in #208 are not actual issues.

File: templates/sshd_config.erb (hunk 2 above)
Comment (Phil-Friderici) [inline templates/sshd_config.erb]: good idea

**Comment by Phil-Friderici** [comment]
After your review, I'll add the same logic for the other parameters mentioned in https://github.com/ghoneycutt/puppet-module-ssh/issues/208

**Comment by Phil-Friderici** [comment]
This change does introduce the possibility to use 'unset' as new value for all incompatible Solaris parameters mentioned in https://github.com/ghoneycutt/puppet-module-ssh/issues/208 that are changeable.

@ghoneycutt It does not take care for unparameterized parameters yet. Should I add a note with the other incompatible parameters somewhere to avoid the problems in the future ?

Let me know when I should squash the commit into one.

**Comment by ghoneycutt** [comment]
Documenting the items sure can't hurt and would be appreciated. Could you please add that and squash your commits and I'll get this merged.

**Comment by Phil-Friderici** [comment]
This change does introduce the possibility to use 'unset' as new value for all incompatible Solaris parameters mentioned in https://github.com/ghoneycutt/puppet-module-ssh/issues/208 that are changeable.

@ghoneycutt It does not take care for unparameterized parameters yet. Should I add a note with the other incompatible parameters somewhere to avoid the problems in the future ?

Let me know when I should squash the commit into one.

**Comment by Phil-Friderici** [comment]
Squashed and documented.

Asked @nalyanyam to test the code.
Let's wait for his feedback.

**Comment by Phil-Friderici** [comment]
@nalyanyam did reply positivly:
> Tested the updated ssh module on Solaris and all ok, Works fine by default or with ‘unset’ values.

Please feel free to merge ;)

**Comment by ghoneycutt** [comment]
Thanks @Phil-Friderici @nalyanyam and @redno2

Released in v3.49.1

**Comment by Phil-Friderici** [comment]
Big THANKS to all of you gentlemen :)

**Pull request merged**